apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender_Configuration.cs

```diff
@@ -233,15 +229,28 @@ public int MaxMemoryCount
 
         private string _credentialFile;
         /// <summary>
-        /// The file path of a service account JSON file to use for authenitication.
+        /// The file path of a service account JSON file to use for authentication.
         /// Not necessary if running on GCE or GAE or if the GOOGLE_APPLICATION_CREDENTIALS environment variable has been set.
+        /// Must not be set if <see cref="CredentialJson"/> is set.
         /// </summary>
         public string CredentialFile
         {
             get => _credentialFile;
             set => _credentialFile = ThrowIfActivated(value, nameof(CredentialFile));
         }
 
+        private string _credentialJson;
+        /// <summary>
+        /// JSON credential for authentication.
+        /// Not necessary if running on GCE or GAE or if the GOOGLE_APPLICATION_CREDENTIALS environment variable has been set.
+        /// Must not be set if <see cref="CredentialFile"/> is set.
+        /// </summary>
+        public string CredentialJson
+        {
+            get => _credentialJson;
+            set => _credentialJson = ThrowIfActivated(value, nameof(CredentialJson));
```

Comment:
Is there a reason `ThrowIfActivated` does not use `[CallerMemberName]`?

Reply:
Prejudice against such magic?